src/watertap_contrib/seto/costing/units/chemical_softening_0D.py

```diff
@@ -0,0 +1,833 @@
+# from watertap.core.zero_order_costing import ZeroOrderCostingData
+# from idaes.core import declare_process_block_class
+
+from pyomo.environ import (
+    Constraint,
+    Expression,
+    Reals,
+    NonNegativeReals,
+    log,
+    Var,
+    Param,
+    Set,
+    Suffix,
+    value,
+    check_optimal_termination,
+    units as pyunits,
+)
+
+from watertap.costing.util import register_costing_parameter_block
+from idaes.core.util.misc import StrEnum
+from ..util import (
+    make_capital_cost_var,
+    make_fixed_operating_cost_var,
+)
+
+"""
+TODO: Remove the hard coded constants in the costing equations
+      Don't fix variables in the unit model code
+"""
+
+
+class SofteningProcedureType(StrEnum):
```

Comment:
You should import this class from the unit model rather than redefining it here.

Reply:
And you can also just test the strings (e.g.`` if softening_type == "single_stage_lime"`` rather than ``if softening_type == SofteningProcedureType.single_stage_lime``)... but I am actually not seeing where this is used anywhere in the costing file so maybe this can be removed?